### PR TITLE
Story1 add new hiring partner

### PR DESCRIPTION
**Pull request by JKapella**
Added ability to register new hiring partners - including: company name, size, tech stack and postcode  as required fields and phone number and company name as optional fields.

Postcode is validated. 

Size options are dynamically set from database.

### CODE REVIEW

**Comment by charliecog** [comment]
url and phone num should be optional

**Comment by charliecog** [comment]
when hp added successfully it should clear the form ready to add another

**Comment by charliecog** [comment]
Hiring partners page - title should say "Add new Hiring partner"

**Comment by charliecog** [comment]
gap between the submit button and the input above (ux)

**Comment by charliecog** [comment]
if you choose company size in drop down and manually change the value of the selected option in the inspector to a number outside of the range it lets you

**Comment by JKapella** [comment]
> url and phone num should be optional - fixed in JS

**Comment by JKapella** [comment]
> when hp added successfully it should clear the form ready to add another

Fixed in JS

**Comment by JKapella** [comment]
> Hiring partners page - title should say "Add new Hiring partner"

Added new sub title in phtml

**Comment by JKapella** [comment]
> gap between the submit button and the input above (ux)

Added fix for this in styles

**Comment by JKapella** [comment]
> if you choose company size in drop down and manually change the value of the selected option in the inspector to a number outside of the range it lets you

Added some front end protection for this in JS

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**(
+     * Sanitise as a string in the hiring_partners_companies table as data.
+     *
+     * @param string $hiringPartnerData
+     *
+     * @return string, which will return the hiring partner data.
+     */
+    public function sanitiseString(string $hiringPartnerData){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            throw new \Exception('An input string does not exist or is too long');
+        }
+    }
+
+    /**
+     * Gets the hiring partner company name
+     *
+     * @return string of company name
+     */
+    public function getCompanyName(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->companyName;
+    }
+
+    /**
+     * Gets the hiring partner company size
+     *
+     * @return string of company size
+     */
+    public function getCompanySize(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->companySize;
+    }
+
+    /**
+     * Gets the hiring partner tech stack
+     *
+     * @return string of tech stack
+     */
+    public function getTechStack(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->techStack;
+    }
+
+    /**
+     * Gets the hiring partner postcode
+     *
+     * @return string of company postcode
+     */
+    public function getPostcode(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->postcode;
+    }
+
+    /**
+     * Gets the hiring partner phone number
+     *
+     * @return string of phone number
+     */
+    public function getPhoneNumber(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Entities/HiringPartnerEntity.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->phoneNumber;
+    }
+
+    /**
+     * Gets the hiring partner website URL
+     *
+     * @return string of website url
+     */
+    public function getWebsiteURL(){
```
Comment (charliecog) [inline src/Classes/Entities/HiringPartnerEntity.php]: return type hinting

File: src/Classes/Factories/CreateHiringPartnerControllerFactory.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+use Psr\Container\ContainerInterface;
+
+class CreateHiringPartnerControllerFactory
+{
+    /**Controller for adding hiring partners to the database
+     *
+     * @param ContainerInterface $container
+     * @return CreateHiringPartnerController instantiated with both params from the container
+     */
+    public function __invoke(ContainerInterface $container)
```
Comment (charliecog) [inline src/Classes/Factories/CreateHiringPartnerControllerFactory.php]: return type hinting

File: src/Classes/Factories/HiringPartnerModelFactory.php [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+class HiringPartnerModelFactory
+{
+    /**
+     * Factory that generate the hiring partner model
+     *
+     * @param ContainerInterface $container
+     *
+     * @return HiringPartnerModel
+     */
+    public function __invoke(ContainerInterface $container)
```
Comment (charliecog) [inline src/Classes/Factories/HiringPartnerModelFactory.php]: return type hinting

**Comment by charliecog** [comment]
optional fields are populating the db with empty string, not null

**Pull request merged**